### ISSUE TITLE
Token decimals were not correct.

### ISSUE THREAD

**Issue by lolieatapple** [suiet/suiet#241]
See my picture:

![image](https://github.com/user-attachments/assets/03524d40-e5fb-4a14-9777-7c7e535db7b2)

I have only 0.001095 USDC (on SUI testnet), but it show 1095 USDC.

in SUI Wallet, it was correct. so it is an issue for Suiet wallet.

![image](https://github.com/user-attachments/assets/8889e96c-143c-4417-946b-13e6d44fc8a7)

**Comment by bruceeewong** [comment]
![image](https://github.com/user-attachments/assets/6ce1744a-3b73-4090-aa4b-2b259a57a38f)

I noticed that your screenshot seems to show portfolio of two difference accounts. Can you double check on the same account if it has this decimal issue?

**Comment by lolieatapple** [comment]
yes, I have two account, but I never have such many USDC, I only have  < 0.1 USDC on the blockchain. 1095 is surely to be decimals error, I can paste my address here, you can query it on testnet explorer.

0x702e89e1070f9a9350b3e6f98d15a59edefcf33ab71581e38c96270e423693cd

**Comment by lolieatapple** [comment]
<img width="362" alt="image" src="https://github.com/user-attachments/assets/9bba095f-a06d-4e78-bbb5-a3b2f29e5542">
This account's USDC balance was wrong, too.

**Comment by Alexahalid** [comment]
it seems there was an issue with the token decimals has it been solved?
@lolieatapple

**Comment by lolieatapple** [comment]
yes, solved.

**Comment by lolieatapple** [comment]
![image](https://github.com/user-attachments/assets/26683990-51f8-4565-ab6e-d63254474e9d)

broken again.

**Comment by bruceeewong** [comment]
This bug is caused due to data indexing issues, and it has been solved.
